pyuvdata/uvdata.py

```diff
@@ -1550,19 +1596,67 @@ def __add__(self, other, run_check=True, check_extra=True,
         ValueError
             If other is not a UVData object, self and other are not compatible
             or if data in self and other overlap.
+
         """
         if inplace:
             this = self
         else:
             this = copy.deepcopy(self)
 
         # Check that both objects are UVData and valid
-        this.check(check_extra=check_extra, run_check_acceptability=run_check_acceptability)
+        this.check(check_extra=check_extra,
+                   run_check_acceptability=run_check_acceptability)
         if not issubclass(other.__class__, this.__class__):
             if not issubclass(this.__class__, other.__class__):
                 raise ValueError('Only UVData (or subclass) objects can be '
                                  'added to a UVData (or subclass) object')
-        other.check(check_extra=check_extra, run_check_acceptability=run_check_acceptability)
+        other.check(check_extra=check_extra,
+                    run_check_acceptability=run_check_acceptability)
+
+        if phase_center_radec is not None and unphase_to_drift:
+            raise ValueError('phase_center_radec cannot be set if '
+                             'unphase_to_drift is True.')
+
+        if unphase_to_drift:
+            if (this.phase_type != 'drift'):
+                warnings.warn("Unphasing this UVData object to drift")
+                this.unphase_to_drift(phase_frame=orig_phase_frame,
+                                      use_ant_pos=use_ant_pos)
+
+            if (other.phase_type != 'drift'):
+                warnings.warn("Unphasing other UVData object to drift")
+                other.unphase_to_drift(phase_frame=orig_phase_frame,
+                                       use_ant_pos=use_ant_pos)
+
+        if phase_center_radec is not None:
+            if np.array(phase_center_radec).size != 2:
+                raise ValueError('phase_center_radec should have length 2.')
+
+            if (this.phase_type == 'drift'
```

Comment:
Thanks very much for changing these comparisons to use `np.isclose`! While I think that's the correct approach, it now makes the `if` clauses hard to parse. Maybe we can add an in-line comment here and above the other clauses explaining that we're checking that the phase centers are the same?

Reply:
sure!